### ISSUE TITLE
[gettext] build block while configuring

### ISSUE THREAD

**Issue by asuper0** [microsoft/vcpkg#26160]
**Host Environment**
- OS: [Windows 10]
- Compiler: VS 2022 x64

**To Reproduce**
default triplet set to x64-windows
`./vcpkg install gettext`

**Failure logs**
[config.log](https://github.com/microsoft/vcpkg/files/9257917/config.log)
[config-x64-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/9257920/config-x64-windows-rel-out.log)
[stdout-x64-windows.log](https://github.com/microsoft/vcpkg/files/9257921/stdout-x64-windows.log)

**Additional context**
It is just blocked while configuring. The log file shows the script is doing "checking for a transitional chinese locale" but can't finished.

**Comment by FrankXie05** [comment]
@asuper0 Please install and switch to English language pack in vs installer.

**Comment by asuper0** [comment]
I already have the English pack. Do you mean change the language in VS->Tools->Options->Environment->International Settings by "switch to English language pack" ? I tried it but blocked again.

BTW, the log file shows some other language can be found:

checking for a traditional french locale... French_France.1252
checking for a traditional japanese locale... none
checking for a french Unicode locale... French_France.65001
checking for a transitional chinese locale...

**Comment by FrankXie05** [comment]
Please provide the error logs :
````
config-x64-windows-dbg-err.log
config-x64-windows-rel-err.log
````

**Comment by asuper0** [comment]
[config-x64-windows-rel-err.log](https://github.com/microsoft/vcpkg/files/9266347/config-x64-windows-rel-err.log)
[config-x64-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/9266350/config-x64-windows-rel-out.log)

I disabled debug in the triplet by "set(VCPKG_BUILD_TYPE release)", so no dbg files.

**Comment by FrankXie05** [comment]
@asuper0 Can you give me a screenshot of the console output after you enter `./vcpkg install gettext`? :)
Like: 
````
PS E:\port-update\vcpkg> ./vcpkg install gettext                                                                       
 Computing installation plan...                                                                                        
 The following packages will be built and installed:                                                                   
 gettext[core]:x86-windows -> 0.21#9                                                                                
 Detecting compiler hash for triplet x86-windows...                                                                 
 Restored 0 package(s) from C:\Users\test\AppData\Local\vcpkg\archives in 358.7 us. Use --debug to see more details.  
 Installing 1/1 gettext:x86-windows...                                                                                
 Building gettext[core]:x86-windows...                                                                               
 -- Using cached gettext-0.21.tar.gz.                                                                                
 -- Extracting source E:/port-update/vcpkg/downloads/gettext-0.21.tar.gz                                              
 -- Applying patch 0002-Fix-uwp-build.patch                                                                          
 -- Applying patch 0003-Fix-win-unicode-paths.patch                                                                   
 -- Applying patch 0004-Fix-uwp-tools-build.patch                                                                     
 -- Applying patch rel_path.patch                                                                                     
 -- Applying patch android.patch                                                                                       
 -- Applying patch gettext-tools_woe32dll_gettextsrc-exports.c.patch                                                  
 -- Using source at E:/port-update/vcpkg/buildtrees/gettext/src/0.21-66fbad380c.clean                                  
 -- Getting CMake variables for x86-windows                                                                            
 -- Using cached msys-gzip-1.11-1-x86_64.pkg.tar.zst.    
 ...
 ...                                                                                
 -- Using msys root at E:/port-update/vcpkg/downloads/tools/msys2/9a1ec3f33446b195                                    
 -- Configuring x86-windows-dbg                                                                                       
 -- Configuring x86-windows-rel                                                                                       
 -- Building x86-windows-dbg                                                                                          
 -- Installing x86-windows-dbg                                                                                        
 -- Building x86-windows-rel                                                                                          
 -- Installing x86-windows-rel     
 ...
 ...                                                                                   
 -- Using cached msys-gzip-1.11-1-x86_64.pkg.tar.zst.                                                                                                                                                                            
 -- Using msys root at E:/port-update/vcpkg/downloads/tools/msys2/79d5b6b790550399                                   
 -- Installing: E:/port-update/vcpkg/packages/gettext_x86-windows/tools/gettext/bin/autopoint                        
 -- Installing: E:/port-update/vcpkg/packages/gettext_x86-windows/share/gettext/gettext/archive.dir.tar.gz           
 -- Performing post-build validation                                                                                 
 -- Performing post-build validation done                                                                            
 Stored binary cache: 'C:\Users\test\AppData\Local\vcpkg\archives\cb\cb0cb9c6359ba856aae5f68825145d7601d1cd5b58698b2d9d8376e062397697.zip'
 Elapsed time to handle gettext:x86-windows: 24.22 min                                                                                         
 Total elapsed time: 24.33 min                                                                                                        
 
PS E:\port-update\vcpkg>
````

**Comment by asuper0** [comment]
For my VCPKG_DEFAULT_TRIPLET=x64-windows, I tried x86 type. It takes a lot of time to build the libiconv ...

[config-x86-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/9267361/config-x86-windows-dbg-err.log)
[stdout-x86-windows.log](https://github.com/microsoft/vcpkg/files/9267363/stdout-x86-windows.log)
[config-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/9267364/config-x86-windows-dbg-out.log)
.

Here is the console output:

PS C:\Users\Administrator> vcpkg install gettext:x86-windows
Computing installation plan...
The following packages will be built and installed:
    gettext[core]:x86-windows -> 0.21#9
  * libiconv[core]:x86-windows -> 1.17
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x86-windows...
-- Automatically setting HTTP(S)_PROXY environment variables to "127.0.0.1:4780".
Restored 0 package(s) from C:\Users\Administrator\AppData\Local\vcpkg\archives in 834.1 us. Use --debug to see more details.
Installing 1/2 libiconv:x86-windows...
Building libiconv[core]:x86-windows...
-- Downloading https://ftp.gnu.org/gnu/libiconv/libiconv-1.17.tar.gz;https://www.mirrorservice.org/sites/ftp.gnu.org/gnu/libiconv/libiconv-1.17.tar.gz -> libiconv-1.17.tar.gz...
-- Extracting source D:/Libs/vcpkg/downloads/libiconv-1.17.tar.gz
-- Applying patch 0002-Config-for-MSVC.patch
-- Applying patch 0003-Add-export.patch
-- Applying patch 0004-ModuleFileName.patch
-- Using source at D:/Libs/vcpkg/buildtrees/libiconv/src/1.17-e2a53830aa.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x86-windows
-- Using cached msys-gzip-1.11-1-x86_64.pkg.tar.zst.
-- Using cached msys-bash-5.1.008-1-x86_64.pkg.tar.zst.
-- Using cached msys-autoconf-2.71-3-any.pkg.tar.zst.
-- Using cached msys-diffutils-3.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-binutils-2.37-5-x86_64.pkg.tar.zst.
-- Using cached msys-libtool-2.4.6-9-x86_64.pkg.tar.xz.
-- Using cached msys-file-5.41-2-x86_64.pkg.tar.zst.
-- Using cached msys-zlib-1.2.11-1-x86_64.pkg.tar.xz.
-- Using cached msys-libbz2-1.0.8-3-x86_64.pkg.tar.zst.
-- Using cached msys-coreutils-8.32-2-x86_64.pkg.tar.zst.
-- Using cached msys-grep-3.0-2-x86_64.pkg.tar.xz.
-- Using cached msys-sed-4.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-libpcre-8.45-1-x86_64.pkg.tar.zst.
-- Using cached msys-m4-1.4.19-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake-wrapper-11-4-any.pkg.tar.zst.
-- Using cached msys-gawk-5.1.0-2-x86_64.pkg.tar.zst.
-- Using cached msys-mpfr-4.1.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-gmp-6.2.1-1-x86_64.pkg.tar.zst.
-- Using cached msys-libreadline-8.1.001-1-x86_64.pkg.tar.zst.
-- Using cached msys-ncurses-6.2-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake1.16-1.16.3-3-any.pkg.tar.zst.
-- Using cached msys-perl-5.32.1-2-x86_64.pkg.tar.zst.
-- Using cached msys-libcrypt-2.1-3-x86_64.pkg.tar.zst.
-- Using cached msys-pkg-config-0.29.2-4-x86_64.pkg.tar.zst.
-- Using cached msys-make-4.3-3-x86_64.pkg.tar.zst.
-- Using cached msys-findutils-4.8.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-libintl-0.21-1-x86_64.pkg.tar.zst.
-- Using cached msys-libiconv-1.16-2-x86_64.pkg.tar.zst.
-- Using cached msys-gcc-libs-11.2.0-3-x86_64.pkg.tar.zst.
-- Using cached msys-msys2-runtime-3.2.0-8-x86_64.pkg.tar.zst.
-- Using msys root at D:/Libs/vcpkg/downloads/tools/msys2/7e05e7aa09f1709f
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at D:/Libs/vcpkg/downloads/tools/msys2/9a1ec3f33446b195
-- Configuring x86-windows-dbg
-- Configuring x86-windows-rel
-- Building x86-windows-dbg
-- Installing x86-windows-dbg
-- Building x86-windows-rel
-- Installing x86-windows-rel
-- Installing: D:/Libs/vcpkg/packages/libiconv_x86-windows/share/libiconv/usage
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: "C:\Users\Administrator\AppData\Local\vcpkg\archives\af\afdc6f5a4ccd797ba2fbcff9515e4b427c980e2e4f4493c3c6e7dd7004fdd9b1.zip"
Elapsed time to handle libiconv:x86-windows: 46.65 min
Installing 2/2 gettext:x86-windows...
Building gettext[core]:x86-windows...
-- Using cached gettext-0.21.tar.gz.
-- Cleaning sources at D:/Libs/vcpkg/buildtrees/gettext/src/0.21-66fbad380c.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/Libs/vcpkg/downloads/gettext-0.21.tar.gz
-- Applying patch 0002-Fix-uwp-build.patch
-- Applying patch 0003-Fix-win-unicode-paths.patch
-- Applying patch 0004-Fix-uwp-tools-build.patch
-- Applying patch rel_path.patch
-- Applying patch android.patch
-- Applying patch gettext-tools_woe32dll_gettextsrc-exports.c.patch
-- Using source at D:/Libs/vcpkg/buildtrees/gettext/src/0.21-66fbad380c.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x86-windows
-- Using cached msys-gzip-1.11-1-x86_64.pkg.tar.zst.
-- Using cached msys-bash-5.1.008-1-x86_64.pkg.tar.zst.
-- Using cached msys-autoconf-2.71-3-any.pkg.tar.zst.
-- Using cached msys-diffutils-3.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-binutils-2.37-5-x86_64.pkg.tar.zst.
-- Using cached msys-libtool-2.4.6-9-x86_64.pkg.tar.xz.
-- Using cached msys-file-5.41-2-x86_64.pkg.tar.zst.
-- Using cached msys-zlib-1.2.11-1-x86_64.pkg.tar.xz.
-- Using cached msys-libbz2-1.0.8-3-x86_64.pkg.tar.zst.
-- Using cached msys-coreutils-8.32-2-x86_64.pkg.tar.zst.
-- Using cached msys-grep-3.0-2-x86_64.pkg.tar.xz.
-- Using cached msys-sed-4.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-libpcre-8.45-1-x86_64.pkg.tar.zst.
-- Using cached msys-m4-1.4.19-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake-wrapper-11-4-any.pkg.tar.zst.
-- Using cached msys-gawk-5.1.0-2-x86_64.pkg.tar.zst.
-- Using cached msys-mpfr-4.1.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-gmp-6.2.1-1-x86_64.pkg.tar.zst.
-- Using cached msys-libreadline-8.1.001-1-x86_64.pkg.tar.zst.
-- Using cached msys-ncurses-6.2-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake1.16-1.16.3-3-any.pkg.tar.zst.
-- Using cached msys-perl-5.32.1-2-x86_64.pkg.tar.zst.
-- Using cached msys-libcrypt-2.1-3-x86_64.pkg.tar.zst.
-- Using cached msys-pkg-config-0.29.2-4-x86_64.pkg.tar.zst.
-- Using cached msys-make-4.3-3-x86_64.pkg.tar.zst.
-- Using cached msys-findutils-4.8.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-libintl-0.21-1-x86_64.pkg.tar.zst.
-- Using cached msys-libiconv-1.16-2-x86_64.pkg.tar.zst.
-- Using cached msys-gcc-libs-11.2.0-3-x86_64.pkg.tar.zst.
-- Using cached msys-msys2-runtime-3.2.0-8-x86_64.pkg.tar.zst.
-- Using msys root at D:/Libs/vcpkg/downloads/tools/msys2/7e05e7aa09f1709f
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at D:/Libs/vcpkg/downloads/tools/msys2/9a1ec3f33446b195
-- Configuring x86-windows-dbg

**Comment by FrankXie05** [comment]
@asuper0 Thank you very much for the information, but from the log you did not complete the debug configuration. 
Is this stuck? Or some other reason? Apparently there is no error message and no broken configuration.

**Comment by asuper0** [comment]
Yes, it stuck every time when  "checking for a transitional chinese locale... ".
I can't find any reason. Tried to stop it and  succeed to install other package.

**Comment by JackBoosY** [comment]
This is **NOT** related to the language pack I think.
It seems that the test program `conftest.exe` is blocked by some other process such as anti-virus.

**Comment by asuper0** [comment]
> This is **NOT** related to the language pack I think. It seems that the test program `conftest.exe` is blocked by some other process such as anti-virus.

Yes, the conftest.exe blocked are the progress, even if the anti-virus program is closed.
However, I succeed to install the gettext by kill conftest.exe process when it blocked. Perhaps this issue can be closed.

**Comment by behrisch** [comment]
I can confirm this problem and that killing conftest.exe helps

**Comment by csorvagep** [comment]
I had the same, and killing conftest.exe helped. Is it possible to fix this somehow?

**Comment by behrisch** [comment]
@JackBoosY @FrankXie05 Should we rather open a new ticket?

**Comment by asuper0** [comment]
It is hard to reproduce. The problem happens every time on one computer, while it works well on an other.

**Comment by taor-163** [comment]
[config-x64-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/12343798/config-x64-windows-dbg-err.log)

**Comment by taor-163** [comment]
./../src/gettext-0-9998579d2a.clean/gettext-runtime/configure: line 7073: gt_JAVACOMP_DISABLED: command not found

**Comment by taor-163** [comment]
after kill conftest.exe process, I succeed to install too.

**Comment by dedowsdi** [comment]
Meet similar problem, kill conftest.exe works.

**Comment by sfhacker** [comment]
This issue is from 2022!

![image](https://github.com/user-attachments/assets/3c722e67-f51f-46c6-8826-882e492e4405)

**Comment by dc6c** [comment]
> This is **NOT** related to the language pack I think. It seems that the test program `conftest.exe` is blocked by some other process such as anti-virus.

I observed that myself: with A/V active, gettext blocks on that locale check, and only killing conftest lets it complete. With A/V temporarily disabled, gettext install completes with no intervention needed.

**Comment by JackBoosY** [comment]
> I had the same problem and got stuck at the “`Configuring x64-windows-dbg`” step. But after I killed the `conftest.exe` in task manager, I found that the `conftest.exe` in the directory `buildtrees\gettext-libintl\x64-windows-dbg` is also deleted automatically, and the `configure` is still stuck without any progress. What should I do about this?

Caused by anti-virus I guess?

**Comment by shun2wang** [comment]
I can also confirm this problem with conftest.exe things. I would vote for reopening too.
